request_test.go

```diff
@@ -522,31 +523,122 @@ func TestRequest_URLConcatenate(t *testing.T) {
 		Reporter: reporter,
 	}
 
-	reqs := []*Request{
-		NewRequestC(config2, "GET", "path"),
-		NewRequestC(config2, "GET", "/path"),
-		NewRequestC(config3, "GET", "path"),
-		NewRequestC(config3, "GET", "/path"),
-		NewRequestC(config3, "GET", "{arg}", "/path"),
-		NewRequestC(config3, "GET", "{arg}").WithPath("arg", "/path"),
-	}
+	const (
+		pathArgType1 = iota
+		pathArgType2 = iota
+	)
 
-	for _, req := range reqs {
-		req.Expect().chain.assertNotFailed(t)
-		assert.Equal(t, "http://example.com/path", client.req.URL.String())
+	requests := []struct {
+		name        string
+		config      Config
+		method      string
+		path        string
+		pathArg     string
+		pathArgType int
+		fullURL     string
+	}{
+		{
+			name:    "config1_empty_url_and_empty_path",
```

Comment:
In most tests we don't use underscores, instead we mostly use spaces and commas, like:

```
empty url, empty path
```

Reply:
Same for other tests.